solutions/USACO dec15-feast-gold.mdx

```diff
@@ -47,3 +47,60 @@ int main() {
 </CPPSection>
 
 </LanguageSection>
+
+## Solution 2 - Using BFS
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int maxn = 5e6 + 5;
+bool pass[maxn];
+
+struct node{
+	int val; bool drunk;
+	node(int _v, bool _d){
+		val = _v;
+		drunk = _d;
+	}
+};
+
+int bfs(int t, int a, int b) {
+	int res = INT_MIN;
+	queue<node> q;
+	q.push({0, 0});
+	while (!q.empty()) {
+		int value = q.front().val;
+		bool drunk = q.front().drunk;
+		q.pop();
+		res = max(res, value);
+		if(value + a <= t && !pass[value + a]) {
+			q.push({value + a, drunk});
+			pass[value + a] = true;
+		}
+		if(value + b <= t && !pass[value + b]) {
+			q.push({value + b, drunk});
+			pass[value + b] = true;
+		}
+		if(!drunk && !pass[value / 2]) {
+			q.push({value / 2, 1});
+			pass[value / 2] = true;

```

Comment:
It's not clear why this is correct, as this might result in `{value/2, 0}` never being added to the queue.

Reply:
yeah, i think its safer to use `bool pass[maxn][2];` where the second dimension indicates if Bessie drank the water or not.

Reply:
So should I change it to `bool pass[maxn][2]` or should I keep it as before?

Reply:
You should change it